### PR TITLE
Create guides landing page

### PR DESCRIPTION
**Pull request by nick-mon1**
## Summary
Adds the landing page for the guides section.

<details><summary>Screenshot</summary>
<p>

Guides _ Digital_gov · 1 01pm · 06-28

</p>
</details> 

### Preview

[Link to Preview](https://federalist-466b7d92-5da1-4208-974f-d61fd4348571.sites.pages.cloud.gov/preview/gsa/digitalgov.gov/nl-create-guides-page/guides/)

### Solution

- adds a landing page for all top level guides (does not display HCD sub-guides)
- uses same `usa-card` component code from the HCD guides for consistency
- adds `dg-guides` and `dg-guides-card` component classes for displaying each guide
- adds front matter field `guide_weight` to each guide to display ASC order (lowest to highest) only on the guides landing page

### How To Test

1. Visit [/guides](https://federalist-466b7d92-5da1-4208-974f-d61fd4348571.sites.pages.cloud.gov/preview/gsa/digitalgov.gov/nl-create-guides-page/guides/)
2. Should see guides listed in card list
3. Clicking on each button will visit the guide

---

### Content 

@bonnieAcameron Here are some outstanding content related items:

- [x] update the `guide_weight` to change order, 1 will display first
- [x] update title to remove "Guide to..."
- [x] update summary for each guide?

---

### Dev Checklist

- [x] PR has correct labels
- [x] A11y testing (voice over testing, meets WCAG, run axe tools)
- [x] Code is formatted properly

### CODE REVIEW

File: themes/digital.gov/layouts/guides/card-guide.html
Comment (nick-mon1) [inline themes/digital.gov/layouts/guides/card-guide.html:5]: @mejiaj I found a way to make the entire card a link but it requires an `<a>` tag that isn't semantic. USWDS documentation for `usa-card` doesn't provide any guidance on this pattern.

File: themes/digital.gov/layouts/guides/list.html
Comment (nick-mon1) [inline themes/digital.gov/layouts/guides/list.html:25]: We use the `data/guidenav` `yaml` files to populate the landing page. Hugo doesn't provide a convenient way to get the top level guide `_index.md` using the where query.

File: themes/digital.gov/src/scss/new/guides/_guides.scss
Comment (nick-mon1) [inline themes/digital.gov/src/scss/new/guides/_guides.scss:1]: Attempted to match the cards used for the `communities` pages.

File: themes/digital.gov/layouts/guides/card-guide.html
Comment (nick-mon1) [inline themes/digital.gov/layouts/guides/card-guide.html:1]: Attempted to match the markup for `usa-card` as close as possible. `usa-card` would require further refactoring to use for this page.

File: themes/digital.gov/src/scss/new/styles.scss
Comment (clmedders) [inline themes/digital.gov/src/scss/new/styles.scss:131]: i wonder if this should be guides_landing to be more descriptive

**Comment by clmedders** [comment]
@nick-mon1 overall looks good... the cards do seem a bit big i wonder if reducing the size is helpful?  im happy to talk it through it but also is a personal thing and dont think it should hold up the work.

File: themes/digital.gov/layouts/guides/card-guide.html
Comment (mejiaj) [reply themes/digital.gov/layouts/guides/card-guide.html:5]: We can consider this a future enhancement.

Once we take this one we can contribute it back to the design system. You can see an example here [Inclusive components - Redundant click](https://inclusive-components.design/cards/#:~:text=pseudo%2Dcontent%20trick-,The%20redundant%20click%20event,-Alternatively%2C%20we%20can)

File: themes/digital.gov/src/scss/new/styles.scss
Comment (mejiaj) [reply themes/digital.gov/src/scss/new/styles.scss:131]: Agreed, perhaps `guides/guide-landing`.

File: themes/digital.gov/layouts/guides/card-guide.html
Comment (mejiaj) [reply themes/digital.gov/layouts/guides/card-guide.html:1]: Yes, and we should pursue this in the future. Right now there are different types of specific cards. 
https://github.com/search?q=repo:GSA/digitalgov.gov+usa-card+language:HTML&type=code&l=HTML

image


Could we re-use `partials/core/usa-card.html` here? If it requires too much re-work for now, we can  table as a future enhancement. But wanted to callout concern of too many cards.

File: themes/digital.gov/layouts/guides/card-guide.html
Comment (mejiaj) [inline themes/digital.gov/layouts/guides/card-guide.html:1]: There's no comment as to what this component is for, where it's used, or how it's used.

Additionally, which fields are required vs optional?

For example the snippet below will always render an image, even if it doesn't exist.

Example:
```html
<div class="dg-guides-card__media">
    {{- partial "core/img-flexible.html" (dict "Site" .Site "src" .Params.Image "small" "false") -}}
  </div>
```

image

**Review by mejiaj — COMMENTED** [review]
Pages look good. I didn't find any a11y issues when running AXE extension. 

Posted some comments/questions below.

**Comment by bonnieAcameron** [comment]
I noticed that all of the words in the guide cards begin with capital letters. I tried to find where/why this is happening, since the guides themselves aren't formatted this way. The red slashed letters should be lowercase: 

Image 5-15-24 at 10 44 PM

File: themes/digital.gov/layouts/guides/card-guide.html
Comment (nick-mon1) [reply themes/digital.gov/layouts/guides/card-guide.html:1]: Thanks for pointing this out, I've added documentation to this partial to explain it's function and use.

Added conditionals for both `$summary` and `$image` fields because there is a potential for them to be missing unlike `$title`.

**Comment by nick-mon1** [comment]
@bonnieAcameron I've updated the casing for the guide titles in the original markdown files which means that each of the guide landing pages will now reflect the above changes.

Screenshot 2024-05-20 at 2 43 55 PM

File: themes/digital.gov/layouts/guides/card-guide.html
Comment (nick-mon1) [reply themes/digital.gov/layouts/guides/card-guide.html:5]: Will create card in the backlog to capture this for additional card refactoring. It's all ready implemented for `news` and `event` cards.

File: themes/digital.gov/layouts/guides/card-guide.html
Comment (nick-mon1) [reply themes/digital.gov/layouts/guides/card-guide.html:1]: I think we should consider adding it to the other refactoring cards. It will take some time to get these in order, one uses `flexbox` and the other `grid` with different positions for the graphics. I've created a card in trello capturing this. 

Screenshot 2024-05-20 at 2 52 56 PM

**Review by bonnieAcameron — CHANGES_REQUESTED** [review]
Can we also adjust the padding on the dg-guides.usa.section to .5rem padding on the top? The space there is a little large.

**Comment by sarah-sch** [comment]
> @bonnieAcameron I've updated the casing for the guide titles in the original markdown files which means that each of the guide landing pages will now reflect the above changes.
> 
> Screenshot 2024-05-20 at 2 43 55 PM

@bonnieAcameron should "human-centered design" also be lowercase in the title for that guide? I would also recommend: "Accessibility for teams"

Also, just a personal peference, but i would recommend replacing the second "two" with "and in the "Understanding robotic process automation" blurb:
Configure bots to execute repetitive tasks **and** save users from performing mundane tasks repeatedly for the same process.

cc @nick-mon1

**Review by sarah-sch — COMMENTED** [review]
I have a capitalization question that we should run by @bonnieAcameron : Should "human-centered design" also be lowercase in the title for that guide? I would also recommend: "Accessibility for teams"

Also, just a personal peference, but i would recommend replacing the second "to" with "and in the "Understanding robotic process automation" blurb:
Configure bots to execute repetitive tasks and save users from performing mundane tasks repeatedly for the same process.

cc @bonnieAcameron

File: content/guides/hcd/introduction/_index.md
Comment (sarah-sch) [inline content/guides/hcd/introduction/_index.md]: "Human-Centered Design" should remain lowercase: "human-centered design"

**Review by sarah-sch — COMMENTED** [review]
Committed a few small text changes for capitalization and to polish intro blurbs.

**Review by bonnieAcameron — COMMENTED** [review]
I'm wondering about the "featured" appearance of the top guides card. Could we stick with a clean grid? In other words, keep the order of the guides but resize the first guide to match the size of all the others? Happy to discuss my reasoning more if needed.

**Comment by nick-mon1** [comment]
@bonnieAcameron I've reset the guide cards to all be the same size.

**Review by bonnieAcameron — COMMENTED** [review]
This looks incredible! the only suggestion I have left is to add padding 20 below the main content. 
Image 6-4-24 at 1 02 PM

**Review by RileySeaburg — DISMISSED** [review]
This is approved pending changes requested by @bonnieAcameron

File: themes/digital.gov/src/scss/new/guides/_guide-landing.scss
Comment (mejiaj) [inline themes/digital.gov/src/scss/new/guides/_guide-landing.scss:5]: Would be better to add these styles when we really need them.

```suggestion
```

File: themes/digital.gov/src/scss/new/guides/_guide-landing.scss
Comment (mejiaj) [inline themes/digital.gov/src/scss/new/guides/_guide-landing.scss:9]:
```suggestion
    display: grid;
    grid-template-columns: repeat(2, 1fr);
```

File: themes/digital.gov/src/scss/new/guides/_guide-landing.scss
Comment (mejiaj) [inline themes/digital.gov/src/scss/new/guides/_guide-landing.scss:18]: Is there a theme color we could use instead of `gray-cool-10`?

File: themes/digital.gov/src/scss/new/guides/_guide-landing.scss
Comment (mejiaj) [inline themes/digital.gov/src/scss/new/guides/_guide-landing.scss:20]: Is this needed? The card is already a single column.

File: themes/digital.gov/src/scss/new/guides/_guide-landing.scss
Comment (mejiaj) [inline themes/digital.gov/src/scss/new/guides/_guide-landing.scss:27]: Theme color token?

File: themes/digital.gov/src/scss/new/guides/_guide-landing.scss
Comment (mejiaj) [inline themes/digital.gov/src/scss/new/guides/_guide-landing.scss:28]: Is this weight used anywhere else? It's more common to see `100`.

Keeping this consistent would allow us to save users from loading another font weight.

**Review by mejiaj — CHANGES_REQUESTED** [review]
A much cleaner design! Added comments below, but looking good.

File: content/guides/accessibility-for-teams/_index.md
Comment (nick-mon1) [reply content/guides/accessibility-for-teams/_index.md:22]: See PR description, this setting allows us to order the guides only on the landing page.

File: content/guides/_index.md
Comment (nick-mon1) [reply content/guides/_index.md:5]: It does nothing in this file, removed.

File: themes/digital.gov/layouts/guides/list.html
Comment (nick-mon1) [reply themes/digital.gov/layouts/guides/list.html:1]: Adding `margin-bottom-5` to be consisten with home page.

File: themes/digital.gov/layouts/guides/list.html
Comment (nick-mon1) [reply themes/digital.gov/layouts/guides/list.html:7]: Yes, these slices are used to extract the guide names and store the guides pages. They need to be declared outside of the range functions to work.

File: themes/digital.gov/src/scss/new/guides/_guide-landing.scss
Comment (nick-mon1) [reply themes/digital.gov/src/scss/new/guides/_guide-landing.scss:47]: Good call, I just noticed the "show all" in the Computed tab was unchecked which is why this was set twice.

File: themes/digital.gov/src/scss/new/guides/_guide-landing.scss
Comment (nick-mon1) [reply themes/digital.gov/src/scss/new/guides/_guide-landing.scss:20]: Yes, removing breaks the layout.

File: themes/digital.gov/layouts/guides/card-guide.html
Comment (nick-mon1) [reply themes/digital.gov/layouts/guides/card-guide.html:1]: Moved back to list.html

File: themes/digital.gov/src/scss/new/guides/_guide-landing.scss
Comment (nick-mon1) [reply themes/digital.gov/src/scss/new/guides/_guide-landing.scss:18]: Yes, replaced it with `base-lightest` which is one step lighter.

File: themes/digital.gov/src/scss/new/guides/_guide-landing.scss
Comment (nick-mon1) [reply themes/digital.gov/src/scss/new/guides/_guide-landing.scss:54]: Using the other style changes, I did not get this to work as expected. We still need the grid and flex styles to make these aligned bottom and consistent for all `card-guides`.

Screenshot 2024-06-20 at 3 43 43 PM

File: content/guides/_index.md
Comment (mejiaj) [inline content/guides/_index.md:6]: Should resources point to `/guides` now?

File: themes/digital.gov/layouts/guides/list.html
Comment (mejiaj) [inline themes/digital.gov/layouts/guides/list.html:37]: ## Future enhancement

Setting a margin on every landing page could be hard to maintain. We'd have the same current problem where you need to update each template manually.

It'd be a better pattern to create a more generic class that holds the page content.

File: themes/digital.gov/layouts/guides/single.html
Comment (mejiaj) [inline themes/digital.gov/layouts/guides/single.html:7]: Remove debug code.

image

File: themes/digital.gov/src/scss/new/guides/_guide-landing.scss
Comment (mejiaj) [inline themes/digital.gov/src/scss/new/guides/_guide-landing.scss:16]: Are there any styles here that can be used across cards?

image

image

image

File: themes/digital.gov/src/scss/new/guides/_guide-landing.scss
Comment (mejiaj) [reply themes/digital.gov/src/scss/new/guides/_guide-landing.scss:51]: Curious, why is the image last? None of the existing cards have it on the bottom.

You don't normally mix grid with flex for this effect. For example, it'd look something like this:

```css
.dg-card {
  display: flex;
  flex-direction: column;
}

.dg-guides-card__content {
  margin-bottom: units(6);
}

.dg-card__media {
  margin-top: auto;
}
```

**Example**
image

**Review by mejiaj — CHANGES_REQUESTED** [review]
Looking good. Minor suggestions and questions.

File: themes/digital.gov/layouts/guides/list.html
Comment (nick-mon1) [reply themes/digital.gov/layouts/guides/list.html:37]: Made a card to capture this.

File: content/guides/_index.md
Comment (nick-mon1) [reply content/guides/_index.md:6]: We don't need it since we now have a `list.html` for the guides.

File: themes/digital.gov/src/scss/new/guides/_guide-landing.scss
Comment (nick-mon1) [reply themes/digital.gov/src/scss/new/guides/_guide-landing.scss:16]: @mejiaj Good call. I've updated the guides card grid to match the HCD card grid. This will reuse existing code, address the style comment below and set us up for refactoring this into a more generic card component.

File: themes/digital.gov/layouts/guides/single.html
Comment (nick-mon1) [reply themes/digital.gov/layouts/guides/single.html:7]: Removed.

**Review by mejiaj — DISMISSED** [review]
LGTM.

**What I tested**
- [x] Mobile and desktop looks good.
- [x] Axe page scan - zero errors found.
- [x] Code style

I also looked at the cloud page build logs and didn't find any impact on performance. Build times look the same and is not impacted by the logic on this template.

**Review by bonnieAcameron — DISMISSED** [review]

**Review by ToniBonittoGSA — APPROVED** [review]
Awesome

**Pull request merged**
